lib/platform.js

```diff
@@ -49,68 +51,115 @@ const PLATFORM_COMMONS = [
 			deviceIdAddress: 0x10000060, // FICR
 			deviceIdPrefix: 'e00fce68'
 		}
-	}
+	},
+	{
+		gen: 3,
+		baseMcu: 'rtl872x',
+		internalFlash: {
+			dfuAltSetting: 0
+		},
+		externalFlash: {
+			dfuAltSetting: 2
+		},
+		dct: {
+			dfuAltSetting: 1,
+			storage: StorageType.FILESYSTEM
+		},
+		filesystem: {
+			storage: StorageType.EXTERNAL_FLASH,
+			address: 0x08600000,
+			size: 2 * 1024 * 1024
+		},
+		encryptedModules: [
+			{
+				// MBR is expected to be encrypted
+				index: 1,
+				type: 'bootloader'
+			}
+		],
+		openOcd: {
+			targetConfig: 'rtl872x.tcl',
+			mcuManufacturer: 'Realtek',
+			deviceIdProcedure: 'rtl872x_read_efuse_mac; rtl872x_wdg_reset',
+			deviceIdPrefix: '0a10aced2021',
+			deviceIdRegex: new RegExp(`MAC:\\s([A-Fa-f0-9]{2}):([A-Fa-f0-9]{2}):([A-Fa-f0-9]{2}):([A-Fa-f0-9]{2}):([A-Fa-f0-9]{2}):([A-Fa-f0-9]{2})`),
+			// FIXME: verification is disabled, it fails on some versions of OpenOCD
+			flashWriteProcedure: (binary, address) => {
+				return `rtl872x_flash_write_bin_ext ${binary} ${address} 1 0`;
+			},
+			resetRunProcedure: 'rtl872x_wdg_reset',
+		}
+	},
 ];
 
-const PLATFORM_COMMONS_BY_GEN = PLATFORM_COMMONS.reduce((map, p) => map.set(p.gen, p), new Map());
+const PLATFORM_COMMONS_BY_MCU = PLATFORM_COMMONS.reduce((map, p) => map.set(p.baseMcu, p), new Map());
 
 // Supported Device OS platforms
 const PLATFORMS = [
 	{
 		id: 6,
 		name: 'photon',
 		displayName: 'Photon',
-		gen: 2
+		gen: 2,
```

Comment:
Let's remove the `gen` property from either `PLATFORMS` or `PLATFORM_COMMONS` definitions now they're matched by MCU name.